File: models/patient.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const patientSchema = new Schema({
    status:{
      type: String,
      enum: ['pending','approved','rejected']
    },
    firstName: {
        type: String,
        minlength: 3,
        maxlength: 35,
        required: true
    },
    lastName: {
        type: String,
        minlength: 3,
        maxlength: 20,
        required: true
    },
    sex:{
      type: String,
      required: true
    },
    address: {
        type: String,
        minlength:5,
        maxlength: 200,
        required: true
    },
    phoneNumber: {
        type: String,
        minlength: 9,
        maxlength: 14,
        required: true
    },
    mail: {
        type: String,
        minlength: 5,
        maxlength:100,
        required: true
    },
    password: {
        type: String,
        minlength: 4,
        maxlength: 4
    }
})

const Patient = mongoose.model('patient', patientSchema)

module.exports = Patient

File: lib/models/patient.dart
class Patient {
  String _id, _firstName, _lastName, _sex, _phoneNumber, _address, _mail;

  Patient(
      {String id,
      String firstName,
      String lastName,
      String sex,
      String phoneNumber,
      String address,
      String mail})
      : _id = id,
        _firstName = firstName,
        _lastName = lastName,
        _sex = sex,
        _phoneNumber = phoneNumber,
        _address = address,
        _mail = mail;

  get mail => _mail;

  set mail(value) {
    _mail = value;
  }

  String get address => _address;

  set address(String value) {
    _address = value;
  }

  String get phoneNumber => _phoneNumber;

  set phoneNumber(String value) {
    _phoneNumber = value;
  }

  String get sex => _sex;

  set sex(String value) {
    _sex = value;
  }

  String get lastName => _lastName;

  set lastName(String value) {
    _lastName = value;
  }

  String get firstName => _firstName;

  set firstName(String value) {
    _firstName = value;
  }

  String get id => _id;

  set id(String value) {
    _id = value;
  }
}
